components/automate-ui/src/app/components/landing/landing.component.ts

```diff
@@ -69,6 +78,8 @@ export class LandingComponent implements OnInit {
       this.router.navigate(this.routeToSegmentList(this.route), { replaceUrl: true });
     } else if (++this.index < this.routePerms.length) {
       this.setPerms();
+    } else {
+        this.onNotFound();
```

Comment:
The loop above this checks each permission in turn. If we run out of permissions to check we will eventually get here. Because the optional `onNotFound` input defaults to a no-op function, no additional `if` is needed here to see if the user supplied it or not.

Reply:
Why is `onNotFound` necessary when there's no callback?  How come you can't just let it pass through without being acted on?

Reply:
`onNotFound` is the callback--but it is an optional parameter. It is set to do nothing by default but callers may override that by providing a value, such as is done here:
https://github.com/chef/automate/blob/07172b329eb8950b6c9b8f82833ac5de803b5b4e/components/automate-ui/src/app/pages/top-nav-landing/top-nav-landing.component.html#L1

Reply:
thanks!